### PR TITLE
Add L-GPU and L-Kokkos as package extras

### PR DESCRIPTION
**Pull request by mlxd**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:** Fixes https://github.com/PennyLaneAI/pennylane/issues/4665 and extends the option to work with Kokkos too.

**Description of the Change:** Re-adds support for `pip install pennylane-lightning[gpu]`. Adds support for `pip install pennylane-lightning[kokkos]`

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Review by AmintorDusko — COMMENTED** [review]
Thank you for that. Could you please update the README file to include the new Lightning-Kokkos installation option?

**Review by vincentmr — APPROVED** [review]
Looks good to me, thanks @mlxd

**Comment by mlxd** [comment]
Just for reference, I can confirm this works locally:

```bash
$~ python -m pip install --extra-index-url http://localhost:8080/simple/ "pennylane-lightning[kokkos]"
Looking in indexes: https://pypi.org/simple, http://localhost:8080/simple/
DEPRECATION: The HTML index page being used (http://localhost:8080/simple/pennylane-lightning/) is not a proper HTML 5 document. This is in violation of PEP 503 which requires these pages to be well-formed HTML 5 documents. Please reach out to the owners of this index page, and ask them to update this index page to a valid HTML 5 document. pip 22.2 will enforce this behaviour change. Discussion can be found at https://github.com/pypa/pip/issues/10825
Collecting pennylane-lightning[kokkos]
  Using cached PennyLane_Lightning-0.32.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (13.6 MB)
WARNING: pennylane-lightning 0.32.0 does not provide the extra 'kokkos'
Collecting pennylane>=0.31
  Using cached PennyLane-0.32.0-py3-none-any.whl (1.5 MB)
Collecting rustworkx
  Using cached rustworkx-0.13.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (2.0 MB)
Collecting numpy<1.24
  Using cached numpy-1.23.5-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (17.1 MB)
Collecting cachetools
  Using cached cachetools-5.3.1-py3-none-any.whl (9.3 kB)
Collecting appdirs
  Using cached appdirs-1.4.4-py2.py3-none-any.whl (9.6 kB)
Collecting autoray>=0.3.1
  Using cached autoray-0.6.6-py3-none-any.whl (54 kB)
Collecting autograd<=1.5
  Using cached autograd-1.5-py3-none-any.whl (48 kB)
Collecting requests
  Using cached requests-2.31.0-py3-none-any.whl (62 kB)
Collecting typing-extensions
  Using cached typing_extensions-4.8.0-py3-none-any.whl (31 kB)
Collecting scipy
  Using cached scipy-1.11.3-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (36.4 MB)
Collecting toml
  Using cached toml-0.10.2-py2.py3-none-any.whl (16 kB)
Collecting semantic-version>=2.7
  Using cached semantic_version-2.10.0-py2.py3-none-any.whl (15 kB)
Collecting networkx
  Using cached networkx-3.1-py3-none-any.whl (2.1 MB)
Collecting future>=0.15.2
  Using cached future-0.18.3-py3-none-any.whl
Collecting idna<4,>=2.5
  Using cached idna-3.4-py3-none-any.whl (61 kB)
Collecting certifi>=2017.4.17
  Using cached certifi-2023.7.22-py3-none-any.whl (158 kB)
Collecting charset-normalizer<4,>=2
  Using cached charset_normalizer-3.3.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (138 kB)
Collecting urllib3<3,>=1.21.1
  Using cached urllib3-2.0.6-py3-none-any.whl (123 kB)
Installing collected packages: appdirs, urllib3, typing-extensions, toml, semantic-version, numpy, networkx, idna, future, charset-normalizer, certifi, cachetools, autoray, scipy, rustworkx, requests, autograd, pennylane-lightning, pennylane
Successfully installed appdirs-1.4.4 autograd-1.5 autoray-0.6.6 cachetools-5.3.1 certifi-2023.7.22 charset-normalizer-3.3.0 future-0.18.3 idna-3.4 networkx-3.1 numpy-1.23.5 pennylane-0.32.0 pennylane-lightning-0.32.0 requests-2.31.0 rustworkx-0.13.2 scipy-1.11.3 semantic-version-2.10.0 toml-0.10.2 typing-extensions-4.8.0 urllib3-2.0.6
```

**Comment by mlxd** [comment]
[sc-47425]

**Review by AmintorDusko — APPROVED** [review]
Thank you for testing locally. Looks good to me!
If you can add the new installation option to README would be great, though.

File: README.rst [hunk 1]
```diff
@@ -190,9 +190,27 @@ Note that OpenMP and BLAS are disabled in this setting.
...
+
+.. code-block:: console
+
+    $ pip install pennylane-lightning[kokkos]
+
+This can be explicitly installed through PyPI as:
+
+.. code-block:: console
+
+    $ pip install pennylane-lightning-kokkos]
```
Comment (AmintorDusko) [inline README.rst:202]:
```suggestion
    $ pip install pennylane-lightning-kokkos
```

**Pull request merged**